stan/math/rev/functor/adj_jac_apply.hpp

```diff
@@ -236,7 +232,7 @@ class adj_jac_vari : public vari {
   }
 
   /**
-   * Implimentation of for_each for applying adjoints.
+   * Implementation of for_each for applying adjoints.
```

Comment:
The docs at the bottom of this file need updated too. In the docs for `adj_jac_apply` the functor interface gets described. That's now outdated: [here](https://github.com/stan-dev/math/blob/8e454a127f7a8753662ced1b3159ce429d599478/stan/math/rev/functor/adj_jac_apply.hpp#L482)

Reply:
Yes good call!

Reply:
Ben can you take another look at these docs? I tried to write an example of what the API looks like

Reply:
I added a couple comments. Looks pretty good!